### DISCUSSION TITLE
Example First Extension not working

### DISCUSSION THREAD

**Discussion by hegdeashwin** [microsoft/vscode-discussions#1281] [Extension Development QnA] [answered]
I'm trying to build a new extension so followed the document https://code.visualstudio.com/api/get-started/your-first-extension
while debugging the default generated code is not working.

Node: v18.13.0
Npm: 8.15.0
VSCode: 1.89.1.   dc96b837cf6bb4af9cd736aa3af08cf8279f7685    arm64
MacOS: Ventura 13.1 (22C65) Apple M1 OR Sonoma 14.5 Apple M1 Max

`yo code` > `Typescript extension` > Webpack
I generated code scaffolding using `yo code`. Selected `TypeScript Extension with Webpack` option.

I also tried below options but same results:

`TypeScript Extension without Webpack`
`TypeScript Extension with unbundled`
`TypeScript Extension with esbuild` => No able to view "Extension Development Host"
`JavaScript Extension`

Followed the below instruction **but its not working**:

```
Get up and running straight away
Press F5 to open a new window with your extension loaded.
Run your command from the command palette by pressing (Ctrl+Shift+P or Cmd+Shift+P on Mac) and typing Hello World.
Set breakpoints in your code inside src/extension.ts to debug your extension.
Find output from your extension in the debug console.
```

<img width="1676" alt="Screenshot 2024-07-20 at 4 05 05 PM" src="https://github.com/user-attachments/assets/6612a7c2-e239-4e68-80a6-f8b6d2512c63">

<img width="1680" alt="Screenshot 2024-07-20 at 4 29 30 PM" src="https://github.com/user-attachments/assets/cde4b57f-9262-4db9-9081-c147683529f1">

Note:

Instruction to execute default dummy test case is working with "Pass" status

```
Run tests
Install the [Extension Test Runner](https://marketplace.visualstudio.com/items?itemName=ms-vscode.extension-test-runner)
Run the "watch" task via the Tasks: Run Task command. Make sure this is running, or tests might not be discovered.
Open the Testing view from the activity bar and click the Run Test" button, or use the hotkey Ctrl/Cmd + ; A
See the output of the test result in the Test Results view.
Make changes to src/test/extension.test.ts or create new test files inside the test folder.
The provided test runner will only consider files matching the name pattern **.test.ts.
You can create folders inside the test folder to structure your tests any way you want.
```

Can anyone please help with the issue? I'm sure how to proceed from here

**Comment by FgrReloaded** [comment]
make sure you have added those commands in your package.json file

**Reply by hegdeashwin** [reply]
Here is my package.json. This is generated by yo. I have not made any changes. `./dist` is generated as well using `npm run watch` as well.

Also, tried adding here but no luck
```
"activationEvents": [
    "onCommand:myplugin.helloWorld"
  ],
```

```
{
  "name": "myplugin",
  "displayName": "myplugin",
  "description": "desc",
  "version": "0.0.1",
  "engines": {
    "vscode": "^1.91.0"
  },
  "categories": [
    "Other"
  ],
  "activationEvents": [],
  "main": "./dist/extension.js",
  "contributes": {
    "commands": [
      {
        "command": "myplugin.helloWorld",
        "title": "Hello World"
      }
    ]
  },
  "scripts": {
    "vscode:prepublish": "npm run package",
    "compile": "webpack",
    "watch": "webpack --watch",
    "package": "webpack --mode production --devtool hidden-source-map",
    "compile-tests": "tsc -p . --outDir out",
    "watch-tests": "tsc -p . -w --outDir out",
    "pretest": "npm run compile-tests && npm run compile && npm run lint",
    "lint": "eslint src --ext ts",
    "test": "vscode-test"
  },
  "devDependencies": {
    "@types/vscode": "^1.91.0",
    "@types/mocha": "^10.0.7",
    "@types/node": "20.x",
    "@typescript-eslint/eslint-plugin": "^7.14.1",
    "@typescript-eslint/parser": "^7.11.0",
    "eslint": "^8.57.0",
    "typescript": "^5.4.5",
    "ts-loader": "^9.5.1",
    "webpack": "^5.92.1",
    "webpack-cli": "^5.1.4",
    "@vscode/test-cli": "^0.0.9",
    "@vscode/test-electron": "^2.4.0"
  }
}

```

**Reply by FgrReloaded** [reply]
can you please share your extension.ts file

**Reply by hegdeashwin** [reply]
```
// The module 'vscode' contains the VS Code extensibility API
// Import the module and reference it with the alias vscode in your code below
import * as vscode from 'vscode';

// This method is called when your extension is activated
// Your extension is activated the very first time the command is executed
export function activate(context: vscode.ExtensionContext) {

	// Use the console to output diagnostic information (console.log) and errors (console.error)
	// This line of code will only be executed once when your extension is activated
	console.log('Congratulations, your extension "myplugin" is now active!');

	// The command has been defined in the package.json file
	// Now provide the implementation of the command with registerCommand
	// The commandId parameter must match the command field in package.json
	const disposable = vscode.commands.registerCommand('myplugin.helloWorld', () => {
		// The code you place here will be executed every time your command is executed
		// Display a message box to the user
		vscode.window.showInformationMessage('Hello World from myplugin!');
	});

	context.subscriptions.push(disposable);
}

// This method is called when your extension is deactivated
export function deactivate() {}
```

**Reply by hegdeashwin** [reply]
I guess If possible you can try generating scaffolding using `npx --package yo --package generator-code -- yo code` and run it

**Reply by FgrReloaded** [reply]
can you create a normal file and then check if command appears ?

**Reply by hegdeashwin** [reply]
Inside "Extension Development Host" created a normal file and tried but no luck

**Reply by FgrReloaded** [reply]
> I guess If possible you can try generating scaffolding using `npx --package yo --package generator-code -- yo code` and run it

I tried and it work for me,
in your debug console are you getting any thing informative ?

**Reply by hegdeashwin** [reply, marked as answer]
Oh! found the issue now. Here is what happening

<img width="711" alt="Screenshot 2024-07-20 at 5 35 43 PM" src="https://github.com/user-attachments/assets/c188dfc6-cbb1-4e01-af72-6dec51cd206c">

After upgrading VSCode to 1.91.1. I'm able to see `Hello World` command and pop up as well.

<img width="540" alt="Screenshot 2024-07-20 at 5 36 49 PM" src="https://github.com/user-attachments/assets/de96b259-2448-4371-8bba-f14d12d67747">

<img width="975" alt="Screenshot 2024-07-20 at 5 37 00 PM" src="https://github.com/user-attachments/assets/7e5977cb-171b-44e8-a0de-727ff1233624">

Thanks a lot @FgrReloaded for help.
